src/Workspaces/Core/Portable/Diagnostics/DiagnosticDataLocation.cs

```diff
@@ -13,15 +13,22 @@ namespace Microsoft.CodeAnalysis.Diagnostics
     [DataContract]
     internal sealed class DiagnosticDataLocation
     {
+        /// <summary>
+        /// Path to where the diagnostic was originally reported.  May be a path to a document in a project, or the
+        /// project file itself.
+        /// </summary>
         [DataMember(Order = 0)]
-        public readonly DocumentId? DocumentId;
+        public readonly string OriginalFilePath;
 
-        // text can be either given or calculated from original line/column
+        /// <summary>
+        /// Document the diagnostic is associated with.  May be null if this is a project diagnostic.
+        /// </summary>
         [DataMember(Order = 1)]
-        public readonly TextSpan? SourceSpan;
+        public readonly DocumentId? DocumentId;
 
+        // text can be either given or calculated from original line/column
         [DataMember(Order = 2)]
-        public readonly string? OriginalFilePath;
+        public readonly TextSpan? SourceSpan;
```

Comment:
this does not really apply to this PR - but I'm not quite clear on the difference between this span and the spans defined by the line/columns.  Why do we need both?

Reply:
yeah, pretty unpleasant.  Ther current reason is that we handle CPS errors and we handle roslyn errors.  The latter have Spans, teh former are the line/col stuff.  And my recollection is that we didn't want to do any up front validation of line/col (esp. as it might refer to documents we don't own).  

This is definitely something to continue cleaning up.  i think in the pull world we only need TextSpan.